src/coreclr/tools/aot/ILCompiler.Compiler/Compiler/ObjectWriter/Dwarf/DwarfFde.cs

```diff
@@ -0,0 +1,113 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Diagnostics;
+using System.Buffers;
+using static ILCompiler.ObjectWriter.DwarfNative;
+
+namespace ILCompiler.ObjectWriter
+{
+    internal sealed class DwarfFde
+    {
+        public readonly DwarfCie Cie;
+        public string PcStartSymbolName;
+        public long PcStartSymbolOffset;
+        public ulong PcLength;
+        public string LsdaSymbolName;
+        public string PersonalitySymbolName;
+        public readonly byte[] Instructions;
+
+        public DwarfFde(DwarfCie cie, byte[] instructions)
+        {
+            Cie = cie;
+            Instructions = instructions;
+            PersonalitySymbolName = null;
+        }
+
+        private enum CFI_OPCODE
+        {
+            CFI_ADJUST_CFA_OFFSET,    // Offset is adjusted relative to the current one.
+            CFI_DEF_CFA_REGISTER,     // New register is used to compute CFA
+            CFI_REL_OFFSET,           // Register is saved at offset from the current CFA
+            CFI_DEF_CFA               // Take address from register and add offset to it.
+        }
+
+        /// <summary>
+        /// Convert JIT version of CFI blob into the the DWARF byte code form.
+        /// </summary>
+        public static byte[] CfiCodeToInstructions(DwarfCie cie, byte[] blobData)
+        {
```

Comment:
It feels a bit fragile that this takes `cie`, and then the result ispassed together with `cie` to the `DwarfFde`constructor and `cie` must match. 
It should be simpler to just pass the CFI instruction blob to the `DwarfFde` constructor and that could transform CFI into instructions using this helper, which could be private as noone else would need it.

Reply:
Agreed. That's a leftover from earlier experiment.